CoreFoundation/Base.subproj/CFPlatform.c

```diff
@@ -175,6 +181,50 @@ const char *_CFProcessPath(void) {
     }
     return __CFProcessPath;
 #else // TARGET_OS_BSD
+    char *argv0 = NULL;
+
+    // Get argv[0].
+#if defined(__OpenBSD__)
+    int mib[2] = {CTL_VM, VM_PSSTRINGS};
+    struct _ps_strings _ps;
+    size_t len = sizeof(_ps);
+
+    if (sysctl(mib, 2, &_ps, &len, NULL, 0) != -1) {
+        struct ps_strings *ps = _ps.val;
+        char *res = realpath(ps->ps_argvstr[0], NULL);
+        argv0 = res? res: strdup(ps->ps_argvstr[0]);
+    }
+#endif
+
+    if (!__CFProcessIsRestricted() && argv0 && argv0[0] == '/') {
+        _CFSetProgramNameFromPath(argv0);
+        free(argv0);
+        return __CFProcessPath;
+    }
+
+    // Search PATH.
+    if (argv0) {
+      char *paths = getenv("PATH");
+      char *p = NULL;
+      while ((p = strsep(&paths, ":")) != NULL) {
+         char pp[PATH_MAX];
+         snprintf(pp, PATH_MAX, "%s/%s", p, argv0);
+         char *res = realpath(pp, NULL);
+         if (!res) {
+             continue;
+         }
+         if (!__CFProcessIsRestricted() && access(res, X_OK) == 0) {
+             _CFSetProgramNameFromPath(res);
+             free(argv0);
+             free(res);
+             return __CFProcessPath;
+         }
+	 free(res);
+      }
+      free(argv0);
+    }
```

Comment:
Could you just tidy up the formatting in this `if` block so its all consistent at 4 space indent, thanks.

Reply:
Oops -- sorry about that. Fixed.